python/tvm/auto_scheduler/relay_integration.py

```diff
@@ -64,19 +65,27 @@ def call_all_topi_funcs(mod, params, target):
         disabled_pass={"AutoSchedulerLayoutRewrite"},
     ):
         try:
-            opt_mod, _ = relay.optimize(mod, target, params)
+            # TODO(jwfromm) Remove this once AlterOpLayout bug that mutates
+            # source module is fixed. Until then, create a clone.
+            mod_clone = deepcopy(mod)
```

Comment:
This line can be lifted out of the try-catch block so that L79 can be simplified.

Reply:
I think we actually need to have both. The problem is that in the first try we attempt to apply `optimize`, which can mutate the source module. Then if that fails, we try to use `vm.lower`, which again can mutate the source module.

Reply:
Ah I see...that's what you meant by the source module was mutated. Yeah this is definitely a bug to be fixed and this is a reasonable workaround.